PAbishai/Ride-My-Way
# Ft api endpoints

**What does this PR do**
Make API endpoints that create and update data in a database and their tests

**Description of tasks to be completed?**
Get the below endpoints and their tests to work
POST /api/v2/auth/signup 
POST /api/v2/auth/login
POST /api/v2/users/rides
GET /api/v2/rides/<ride_id>
POST /api/v2/rides/<ride_id>/requests
GET /api/v2/users/rides/<ride_id>/requests
PUT /api/v2/users/rides/<ride_id>/requests/<request_id>'

**How should this be manually tested?**
Install postgress
Create a database ride-my-way
`createdb ride-my-way`
Create tables users and rides
```
`CREATE TABLE rides (
id SERIAL PRIMARY KEY,
user_id  INT NOT NULL,
location VARCHAR NOT NULL,
destination VARCHAR NOT NULL,
leaving VARCHAR NOT NULL, passengers VARCHAR)
;
```
```
CREATE TABLE users (
id SERIAL PRIMARY KEY,
name varchar NOT NULL,
email varchar NOT NULL,
dl_path varchar,
car_reg varchar,
password varchar
);
```
Clone the repo
Create a virtualenv 
`#virtualenv Ride-My-Way`
cd into the repo and activate it
run
`pip install -r requirements.txt`
cd into the resources folder
edit 'api_v2_users.py'
add your postgres username and password to conn_string
run 
`python api_v2_users.py`
flask should start
the apis can be tested on localhost:5000
To do the tests
cd out of resources and into tests
run 
`pytest test_apis_v2.py`
the tests should pass
